src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact() {
+        super();
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Address // instanceof handles nulls
```

Comment:
The class is of type Contact and you are checking if it is an instance of Address. It is incorrect.